objectivec/GPBDescriptor.m

```diff
@@ -252,6 +252,49 @@ - (GPBDescriptor *)containingType {
   return [parentClass descriptor];
 }
 
+- (NSString *)fullName {
+  NSString *className = NSStringFromClass(self.messageClass);
+  GPBFileDescriptor *file = self.file;
+  NSString *objcPrefix = file.objcPrefix;
+  if (objcPrefix && ![className hasPrefix:objcPrefix]) {
+    NSAssert(0,
+             @"Class didn't have correct prefix? (%@ - %@)",
+             className, objcPrefix);
+    return nil;
+  }
+  GPBDescriptor *parent = self.containingType;
+
+  NSString *name = nil;
+  if (parent) {
+    NSString *parentClassName = NSStringFromClass(parent.messageClass);
+    // The generator will add _Class to avoid reserved words, drop it.
+    if (objc_getAssociatedObject(parent, &kClassNameHasSuffixKey)) {
+      parentClassName =
+        [parentClassName substringToIndex:(parentClassName.length - 6)];
```

Comment:
Yea, I don't like the fact that it knows what was added but there's other cases like that in things like TextFormat.  I'll see if I can come up with a better model.